### PR TITLE
[FrameworkBundle] Use CPP

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | N/A
| License       | MIT

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/CachePoolPruneCommand.php [hunk 1]
```diff
@@ -26,16 +26,13 @@
...
 {
-    private iterable $pools;
-
     /**
      * @param iterable<mixed, PruneableInterface> $pools
      */
-    public function __construct(iterable $pools)
-    {
+    public function __construct(
+        private iterable $pools
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Command/CachePoolPruneCommand.php]:
```suggestion
        private iterable $pools,
```

File: src/Symfony/Bundle/FrameworkBundle/HttpCache/HttpCache.php [hunk 2]
```diff
@@ -28,19 +28,19 @@
...
      */
-    public function __construct(KernelInterface $kernel, string|StoreInterface|null $cache = null, ?SurrogateInterface $surrogate = null, ?array $options = null)
-    {
-        $this->kernel = $kernel;
-        $this->surrogate = $surrogate;
+    public function __construct(
+        protected KernelInterface $kernel,
+        string|StoreInterface|null $cache = null,
+        private ?SurrogateInterface $surrogate = null,
+        ?array $options = null
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/HttpCache/HttpCache.php]:
```suggestion
        ?array $options = null,
```

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php [hunk 3]
```diff
@@ -41,15 +41,11 @@ class AssetsInstallCommand extends Command
...
     public const METHOD_RELATIVE_SYMLINK = 'relative symlink';
 
-    private Filesystem $filesystem;
-    private string $projectDir;
-
-    public function __construct(Filesystem $filesystem, string $projectDir)
-    {
+    public function __construct(
+        private Filesystem $filesystem,
+        private string $projectDir
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php]: (from fabbot):
```suggestion
        private string $projectDir,
```

File: src/Symfony/Bundle/FrameworkBundle/Command/CachePoolClearCommand.php [hunk 4]
```diff
@@ -32,18 +32,14 @@
...
-    private ?array $poolNames;
-
     /**
      * @param string[]|null $poolNames
      */
-    public function __construct(Psr6CacheClearer $poolClearer, ?array $poolNames = null)
-    {
+    public function __construct(
+        private readonly Psr6CacheClearer $poolClearer,
+        private readonly ?array $poolNames = null,
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Command/CachePoolClearCommand.php]: What is the reason for adding `readonly` here but not in other places?

Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/Command/CachePoolClearCommand.php]: Oh, how did that happen? 😓

**Review by nicolas-grekas — APPROVED** [review]

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**